### ISSUE TITLE
CD failing

### ISSUE THREAD

**Issue by tmcadam** [ACCESS-NRI/ncigrafana#5]
CD is failing for ncigrafana. I suspect due to default sqlalchemy in conda changing from 1.4 to 2.0 in the 6 months since last run. Do I just need to do something like this - sqlalchemy<2.0 in conda/environment.yml to limit the version? I see other references to sqlalchemy in conda/meta.yaml

https://github.com/ACCESS-NRI/ncigrafana/actions/runs/12171650433/job/33948936071

**Comment by aidanheerdegen** [comment]
Yeah it is a pain the number of places that the same information needs to be included to support different packaging systems.

That is why this approach was adopted:

https://github.com/payu-org/payu/blob/master/conda/meta.yaml#L14-L17

which uses `pyproject.toml` and some ninja macro stuff to pull the dependency info from a single source. Note that this requires the version info has a space between the package name and version specification

https://github.com/payu-org/payu/blob/master/pyproject.toml#L25-L34

TL;DR I think it only needs to be changed in `meta.yaml` for install.

**Comment by tmcadam** [comment]
Have fixed the issue in CD workflow. I'll wait until I can test package in NCI before closing off.

**Comment by tmcadam** [comment]
Tested and seems to be working fine
